src/DynamicData/List/ObservableListEx.cs

```diff
@@ -957,6 +957,186 @@ public static IObservable<IEnumerable<T>> LimitSizeTo<T>(this ISourceList<T> sou
     }
 
+    /// <summary>
+    /// Operator similiar to Merge except it is ChangeSet aware.  All of the observable changesets are merged together into a single stream of ChangeSet events.
+    /// </summary>
+    /// <typeparam name="TObject">The type of the object.</typeparam>
+    /// <param name="source">The Source Observable ChangeSet.</param>
+    /// <returns>The result from merging the changesets together.</returns>
+    /// <exception cref="ArgumentNullException">Parameter was null.</exception>
+    public static IObservable<IChangeSet<TObject>> MergeChangeSets<TObject>(this IObservable<IObservable<IChangeSet<TObject>>> source)
+        where TObject : notnull
+    {
+        source.ThrowArgumentNullExceptionIfNull(nameof(source));
+
+        return new MergeChangeSets<TObject>(source, equalityComparer: null).Run();
+    }
+
+    /// <summary>
+    /// Operator similiar to Merge except it is ChangeSet aware.  All of the observable changesets are merged together into a single stream of ChangeSet events.
+    /// </summary>
+    /// <typeparam name="TObject">The type of the object.</typeparam>
+    /// <param name="source">The Source Observable ChangeSet.</param>
+    /// <param name="equalityComparer"><see cref="IEqualityComparer{T}"/> instance to determine if two elements are the same.</param>
+    /// <returns>The result from merging the changesets together.</returns>
+    /// <exception cref="ArgumentNullException">Parameter was null.</exception>
+    public static IObservable<IChangeSet<TObject>> MergeChangeSets<TObject>(this IObservable<IObservable<IChangeSet<TObject>>> source, IEqualityComparer<TObject> equalityComparer)
+        where TObject : notnull
+    {
+        source.ThrowArgumentNullExceptionIfNull(nameof(source));
+        equalityComparer.ThrowArgumentNullExceptionIfNull(nameof(equalityComparer));
+
+        return new MergeChangeSets<TObject>(source, equalityComparer).Run();
+    }
+
+    /// <summary>
+    /// Operator similiar to Merge except it is ChangeSet aware.  Merges both observable changesets into a single stream of ChangeSet events.
+    /// </summary>
+    /// <typeparam name="TObject">The type of the object.</typeparam>
+    /// <param name="source">The Source Observable ChangeSet.</param>
+    /// <param name="other">The Other Observable ChangeSet.</param>
+    /// <param name="scheduler">(Optional) <see cref="IScheduler"/> instance to use when enumerating the collection.</param>
+    /// <param name="completable">Whether or not the result Observable should complete if all the changesets complete.</param>
+    /// <returns>The result from merging the changesets together.</returns>
+    /// <exception cref="ArgumentNullException">Parameter was null.</exception>
+    public static IObservable<IChangeSet<TObject>> MergeChangeSets<TObject>(this IObservable<IChangeSet<TObject>> source, IObservable<IChangeSet<TObject>> other, IScheduler? scheduler = null, bool completable = true)
```

Comment:
`completable` is modeled after `AsObservableChangeSet`.  The scheduler is like `AsObservable` (converts `IEnumerable<T>` to `IObservable<T>`).

All of these functions are like their Cache counterparts, so if we don't add them, Cache and List will be asymmetrical.  We could change both but that would break the new API (which might be fine, it is brand new).

It is true that `IObservable<IObservable<>>` might be the only overload that is truly necessary.  However, directly supporting the IEnumerable case allows the collection to be initialized as a single changeset, as an optimization.

Reply:
I see it as a convenience feature, and in my mind there are two concerns for adding convenience: how complicated the thing you're making convenient is, and how common of a scenario it is.

Both of these conveniences are quite simply, so the distinction for me is how common a scenario it is. Separate overloads for `IEnumerable<>` versus `IObservable<>` I think is plenty common to merit an extra set of overloads, but building a stream that ignores completion notifications I think isn't nearly common enough.

And as I mentioned in another comment, I don't think parity between Cache and List is really a valid goal, there's too much fundamental difference between the two. I'd argue that lists as a _whole_ are really a niche use-case, and the vast majority of workloads should be using Caches.

Anyway, those are just my opinions. I'd defer to @RolandPheasant when it comes to how much feature parity we want to support.